modules/ecs-service/README.md

```diff
@@ -1,23 +1,204 @@
+
+An ECS task definition is organized into families which consist of multiple revisions of a given task definition. Using the blueprint analogy, these are like different versions of the blueprint.
+
+#### Container Definitions
+
+The heart of the task definition is the container definitions. These configurations define the container deployed as part of the ECS task, including the image, environment variables, secrets, and other container settings. Many of these options are similar to docker run flags. You can find detailed information in the [ECS container definition documentation](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definitions) for more information.
+
+##### Essential Container Definition
+
+The essential container definition is the primary container deployed as part of the ECS task and is required for all ECS tasks. The ECS service module supports a single essential container definition. If you need to deploy multiple containers within a task, contact the module maintainers to add support for this feature.
+
+#### Task Execution Role
+
+The ECS task execution role is an IAM role granting permissions to the ECS task during provisioning. It is assigned the `AmazonECSTaskExecutionRolePolicy` policy by default, and additional policies, such as access to Secrets Manager secrets, can be attached using the `ecs_container_secrets` and `docker_credential_secretsmanager_arn` variables.
+
+#### Task Role
+
+The ECS task role is an IAM role granting permissions to the ECS task at runtime. By default, no policies are attached, but you can add policies using the `ecs_task_role_policy_arns` variable. This may be necessary if the ECS task needs to access other AWS services like S3 or SNS.
+
+#### Container Logging
+
+Container logging can be enabled to send container logs to CloudWatch Logs, useful for debugging and monitoring. Use the `enable_container_logs` variable to enable or disable this feature.
+
+#### Container Environment Variables
+
+You can configure non-sensitive environment variables defined in the container definition using the `ecs_container_environment_variables` variable, which is a map of variable names to values. An example of setting environment variables is shown below.
+
+```hcl
+module "example" {
+  # ...
+
+  ecs_container_environment_variables = { 
+    ENV_VAR_1 = "value1"
+    ENV_VAR_2 = "value2"
+  }
+
+  # ...
+}
+```
+
+##### Secrets Manager
+
+Sensitive environment variables can be injected using the ecs_container_secrets variable, which maps variable names to AWS Secrets Manager secret ARNs. **The assumption is made that the name of the secret is the same as the name of the environment variable.** The module automatically handles the injection of secret values into the container at runtime. An example of setting secrets is shown below.
```

Comment:
Is the assumption for the team or for AWS?

Reply:
For the team. I'll specify.